gsd/pygsd.py

```diff
@@ -158,19 +155,16 @@ def __init__(self, file):
         self.__namelist = {}
         c = 0
         self.__file.seek(self.__header.namelist_location, 0)
-        for i in range(self.__header.index_allocated_entries):
-            namelist_entry_raw = \
-                self.__file.read(gsd_namelist_entry_struct.size)
-            n = gsd_namelist_entry._make(
-                gsd_namelist_entry_struct.unpack(namelist_entry_raw))
-
-            # 0 first byte signifies end
-            sname = n.name.rstrip(b'\x00').decode('utf-8')
-            if len(sname) == 0:
-                break
+        namelist_raw = self.__file.read(
+            self.__header.namelist_allocated_entries * 64)
```

Comment:
How does this work for 2.0? Won't you read too little if the names are longer?

Reply:
In 2.0, the size of the name list is `namelist_allocated_entries * 64` so this reads all of the names. This is confusing, but I kept it that way for consistency with v1 files where ``namelist_allocated_entries` is the *number* of slots for 64-byte names. In v2, names can be any length and are separated by a single NULL character in the buffer. In v1, each element name was NULL padded out to 64 bytes.